src/jwplayer/players/CanonicalVideoPlayer/canonicalOnReady.ts

```diff
@@ -0,0 +1,27 @@
+import { communicationService } from 'jwplayer/utils/communication';
+import { willAutoplay, willMute } from 'jwplayer/utils/articleVideo/articleVideoConfig';
+import { recordVideoEvent, VIDEO_RECORD_EVENTS } from 'jwplayer/utils/videoTimingEvents';
+
+export default function canonicalOnReady(videoDetails, playerInstance): void {
+	const playerKey = 'aeJWPlayerKey';
+
+	window.dispatchEvent(new CustomEvent('wikia.jwplayer.instanceReady', { detail: playerInstance }));
```

Comment:
Interesting. I didn't see this part in any of the other players from what I remember.

Reply:
Is this for tracking, or is this for AdEngine?

Reply:
Yup, it exists on [articlePlayerOnReady.ts](https://github.com/Wikia/jwplayer-fandom/blob/master/src/jwplayer/utils/articleVideo/articlePlayerOnReady.ts) for the UCP player too